modules/text_server_adv/text_server_adv.cpp

```diff
@@ -1432,8 +1434,33 @@ _FORCE_INLINE_ bool TextServerAdvanced::_ensure_cache_for_size(FontAdvanced *p_f
 #endif
 
 		if (!p_font_data->face_init) {
-			// Get style flags and name.
-			if (fd->face->family_name != nullptr) {
+			// When a font does not provide a `family_name`, FreeType tries to synthesize one based on other names.
```

Comment:
The same should be added to the `TextServerFallback` as well.

For the `TextServerAdvanced`, we can use probably use [hb_ot_name_get_utf32](https://harfbuzz.github.io/harfbuzz-hb-ot-name.html#hb-ot-name-get-utf32) method (which will simplify the code and probably can handle more encodings). See `TextServerAdvanced::_font_get_ot_name_strings` which is already using it.